### ISSUE TITLE
Floating point parsers not working on macOS Catalina

### ISSUE THREAD

**Issue by maximkrouk** [pointfreeco/swift-parsing#9]
I'll just leave the use case, it should probably be enough to reproduce the issue

```swift
.package(..., from: "0.1.1")
```

```swift
let output = Parsers.StartsWith("v")
    .skip(Many(StartsWith(" ")))
    .take(Int.parser())
    .parse("v  1.0")
print(output) // Optional(1) ✅
```

```swift
let output = Parsers.StartsWith("v")
    .skip(Many(StartsWith(" ")))
    .take(Float.parser())
    .parse("v  1.0")
print(output) // nil 🚫
```

```swift
print(Float.parser().parse("1.0")) // nil 🚫
```

> Double doesn't work as well

**Comment by stephencelis** [comment]
@maximkrouk I can't seem to reproduce this. When I add the following assertion to our test suite, it passes:

```swift
XCTAssertEqual(
  Parsers.StartsWith("v")
    .skip(Many(StartsWith(" ")))
    .take(Float.parser())
    .parse("v  1.0"),
  1.0
)
```

Any chance you can try submitting a failing PR?

**Comment by maximkrouk** [comment]
Does the CI workflow use macOS Catalina for tests? iOS works fine btw, so I'm not sure if I'm able to make a failing PR 🌚

**Comment by stephencelis** [comment]
CI is on Catalina. I ran locally on Big Sur and it's working fine.

**Comment by maximkrouk** [comment]
Well, it's fixed on `main`, but the current `0.1.1` version fails

**Comment by stephencelis** [comment]
@maximkrouk That's really weird. We changed the `Double` parsing strategy, so it makes sense that they _could_ differ, but I ran the above assertion against the `0.1.1` tag. We'll cut 0.1.2 soon!

**Comment by dcaunt** [comment]
FWIW I also ran into this, and can confirm it's fixed in 0.1.2.
